hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java

```diff
@@ -253,6 +253,10 @@
   public static final String COMPACTION_AFTER_BULKLOAD_ENABLE =
       "hbase.compaction.after.bulkload.enable";
 
+  /** Config for allow split when file count greater than the configured blocking file count*/
+  public static final String ENABLE_SPLIT_BEFORE_COMPACT_IGNORE_BLOCKING_KEY =
```

Comment:
On the JIRA you (IMHO) convincingly argue this is a bug:
> But the region's compact priority is the minimum of all the stores, when the number of storefiles in a store is larger than the configed `hbase.hstore.blockingStoreFiles`, the priority will be a negative number, but the compared priority in requestSplit() is 1(PRIORITY_USER).

So should we treat this like a bug fix? Blocking file limit should not prevent splits. If so, there is no need for a new configuration setting here. 
Now it can be argued that splitting a region that has so many store files is suboptimal. That is true. It can be addressed by a follow up issue as an optimization. (Not sure, perhaps split procedure can be modified to do compaction inline in such degenerate cases, but anyway is an orthogonal conversation.)

Reply:
Thanks, @apurtell , I used to consider this issue as a bug. But Duo advised in the above conversation that this is a feature, and a configure is needed for this big behavior change, the original words are as follows,

> I think this is a big behavior change? We used to disable split when store file count is greater than the blocking count, this is feature, not a bug...
> 
> So if you want to change this behavior, you'd better write a design doc or something to better introduce the background, why we should change this behavior, the advantage and disadvantages, etc.
> 
> And if it is not easy to suit all the scenario, is it possible to make it configurable?

I think split here is somewhat needed to prevent region from endless growing unexpectedly, I described a scenario in the issue https://issues.apache.org/jira/browse/HBASE-26242. But it is not required at all the time as you and Duo said above, and in some scenarios it is suboptimal. Currently, can we keep it configurable? WDYT? If you agree, I can change the issue description and clearly the wish for the behavior change. Thanks.

Reply:
@sunhelly Ok, we can keep this configurable to satisfy everyone.